src/Scraper/Word.fs

```diff
@@ -2,40 +2,60 @@
 
 open Article
 
-let recordCzechPartOfSpeech word = function
+let recordCzechPartOfSpeech article = function
     | "podstatné jméno" -> [
-        if word |> NounValidation.isPluralValid
-        then word |> NounRegistration.registerNounPlural
-
-        if word |> NounValidation.isAccusativeValid
-        then word |> NounRegistration.registerNounAccusative
+        article
+        |> NounValidation.parseNounPlural
+        |> Option.map NounRegistration.registerNounPlural
+        |> Option.defaultValue (async { return () })
```

Comment:
this is a bit confusing. is it no operation in case of None? I think this could be some function like NoOperation

Reply:
Yes, correct. I guess this does not read really clear... I refactored things a little bit, take a look